app/api/v1/views.py

```diff
@@ -2,27 +2,37 @@
 from flask import jsonify, make_response, request
 import datetime
 
-from .models import IncidenceModel
+from .models import RedFlagModel
 
+new_flag = RedFlagModel()
 
-class RedFlags(Resource):
+class RedFlags(Resource, RedFlagModel):
     def __init__(self):
-        self.db = IncidenceModel()
-         
+        self.db = RedFlagModel(CreatedBy, Location, Status, Comment)
```

Comment:
This solves the "missing 4 parameters error" and starts raising "CreatedBy not defined", Can you assist me on actionable steps that can solve this? I will also appreciate any resources that can enhance my understanding specifically on OOP scenarios as the one above.